test/src/test/java/jenkins/install/LoadDetachedPluginsTest.java

```diff
@@ -81,6 +84,25 @@ public void upgradeFromJenkins2() {
         });
     }
 
+    @Issue("JENKINS-48604")
+    @Test
+    @LocalData
+    public void upgradeFromJenkins2WithDependency() {
+        VersionNumber since = new VersionNumber("2.0");
+        rr.then((JenkinsRule r) -> {
+            List<DetachedPlugin> detachedPlugins = ClassicPluginStrategy.getDetachedPlugins(since);
+            assertThat("Plugins have been detached since the pre-upgrade version",
+                    detachedPlugins.size(), greaterThan(1));
+            assertThat("Plugins detached between the pre-upgrade version and the current version should be installed",
+                    getInstalledDetachedPlugins(r, detachedPlugins).size(), equalTo(detachedPlugins.size()));
+            Plugin scriptSecurity = r.jenkins.getPlugin("script-security");
+            assertThat("Script-security should be installed", scriptSecurity, notNullValue());
+            assertThat("Dependencies of detached plugins should not be downgraded",
+                    scriptSecurity.getWrapper().getVersionNumber(), greaterThan(new VersionNumber("1.18.1")));
```

Comment:
```java
is(new VersionNumber("2.34"))
```

should in fact pass, right?

Reply:
`1.34` is the version from `@LocalData`, but yes that should pass. Should I assert that instead for clarity?

Reply:
I think `is(new VersionNumber("1.34"))` is better than the current assertion:

This test currently relies on the fact that the version of `command-launcher` in `split-plugins.txt` depends on an older version of  `script-security` than what is bundled in `@LocalData`. If we updated the version of `command-launcher` in `split-plugins.txt` to one with a newer `script-security` dependency (maybe for a security fix of `command-launcher`?), the test would still pass, but for the wrong reason. Using your assertion would cause the test to fail when `split-plugins.txt` was updated in that scenario.